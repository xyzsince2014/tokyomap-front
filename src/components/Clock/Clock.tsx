import {useContext} from 'react';

import DatetimeContext from '../../providers/datetime/datetimeContext';

const Clock: React.FC = () => {
  const {datetime} = useContext(DatetimeContext);
  return <div className="p-clock">{datetime}</div>;
};

export default Clock;
