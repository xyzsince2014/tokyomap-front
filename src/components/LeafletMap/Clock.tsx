import * as React from 'react';

// todo: update the time display every minute
const Clock: React.FC = () => {
  const datetime: string = new Date()
    .toLocaleString('en-GB', {
      // see detail here http://www.htmq.com/js/date_toLocaleString.shtml
      hour12: false,
      day: 'numeric',
      month: 'short',
      hour: 'numeric',
      minute: 'numeric',
    })
    .replace(',', '');

  return <div className="c-clock">{datetime}</div>;
};

export default Clock;
