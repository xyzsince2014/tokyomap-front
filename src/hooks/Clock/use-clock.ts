import {useState, useEffect} from 'react';

import {fetchCurrentDatetime} from '../../utils/dateTime';

const useClock = () => {
  const [datetime, setDatetime] = useState(fetchCurrentDatetime());

  const tick = () => {
    setDatetime(fetchCurrentDatetime());
  };

  useEffect(() => {
    const timerId = setInterval(tick, 1000);
    return () => clearInterval(timerId);
  }, []);

  return datetime;
};

export default useClock;
