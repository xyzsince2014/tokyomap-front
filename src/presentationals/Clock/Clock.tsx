import * as React from 'react';

interface ClockProps {
  datetime: string;
}

const Clock: React.FC<ClockProps> = ({datetime}) => <div className="c-clock">{datetime}</div>;

export default Clock;
