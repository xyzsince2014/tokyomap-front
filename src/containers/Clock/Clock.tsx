import * as React from 'react';

import Clock from '../../components/Clock/Clock';
import {fetchCurrentDatetime} from '../../utils/dateTime';

const useClock = () => {
  const [datetime, setTime] = React.useState<string>(fetchCurrentDatetime());

  const tick = () => {
    setTime(fetchCurrentDatetime());
  };

  React.useEffect(() => {
    const timer = setInterval(tick, 1000);
    return () => clearInterval(timer);
  }, []);

  return datetime;
};

const ClockContainer: React.FC<{}> = () => <Clock datetime={useClock()} />;

export default ClockContainer;
