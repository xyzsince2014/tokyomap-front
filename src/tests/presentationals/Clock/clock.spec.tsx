import React from 'react';
import renderer from 'react-test-renderer';

import Clock from '../../../presentationals/Clock/Clock';

describe('Clock', () => {
  const component = renderer.create(<Clock datetime="hoge" />);
  const clockJson = component.toJSON();

  test('render Clock', () => {
    expect(clockJson).toMatchSnapshot();
  });
});
