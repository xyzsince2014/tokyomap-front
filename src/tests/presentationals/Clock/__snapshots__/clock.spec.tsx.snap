// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock render Clock 1`] = `
<div
  className="c-clock"
>
  hoge
</div>
`;
